src/ShellRouting/IShellRouter.cs

```diff
@@ -26,6 +27,16 @@ public interface IShellHandler
         ///     The incoming message to be handled.
         /// </param>
         public void Handle(Message message);
+
+        /// <summary>
+        ///     Called by the shell server to asynchronously handle a message
+        ///     coming in from the client.
+        /// </summary>
+        /// <param name="message">
+        ///     The incoming message to be handled.
+        /// </param>
+        public async Task HandleAsync(Message message) =>
```

Comment:
Perhaps have a generic return type like Task<MessageResponse<TResponse>>
where the MessageResponse contains some basic properties like Status (OK/Error) and a TResponse type that is the message response to the client.
If the router finds that the message response is not null, it automatically sends it back to the client.

In this case, the router needs to await for the response, and therefore handle the request queue.

Reply:
I went with a slightly different approach, but that might be a good improvement in the future!